providers/irdma/uverbs.c

```diff
@@ -1328,7 +1328,9 @@ struct ibv_qp *irdma_ucreate_qp(struct ibv_pd *pd,
 
 	if (attr->cap.max_send_sge > uk_attrs->max_hw_wq_frags ||
 	    attr->cap.max_recv_sge > uk_attrs->max_hw_wq_frags ||
-	    attr->cap.max_inline_data > uk_attrs->max_hw_inline) {
+	    attr->cap.max_inline_data > uk_attrs->max_hw_inline ||
+	    attr->cap.max_send_wr > IRDMA_MAX_SEND_WR ||
+	    attr->cap.max_recv_wr > IRDMA_MAX_RECV_WR) {
```

Comment:
I am not seeing a need for this.

irdma_get_sqdepth guards against unsupported values of passed in max_send_wr?

Reply:
I was trying to avoid the failure of the following pyverbs tests, The test tries to pass the following values and expects the driver to return EINVAL...

qp_init_attr.max_recv_wr = 0xffffffff # max_uint32
qp_init_attr.max_send_wr = 0xffffffff # max_uint32

```
 ./build/bin/run_tests.py --dev irdma1 test_qp 
ss.F.Fss....ssssss..s.sss.
======================================================================
FAIL: test_create_raw_qp_ex_with_illegal_caps_max_recv_wr (tests.test_qp.QPTest)
Test Raw Packet QP creation via ibv_create_qp_ex with a QPAttr object with illegal max_recv_wr.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/rdma-core/tests/test_qp.py", line 234, in test_create_raw_qp_ex_with_illegal_caps_max_recv_wr
    self.create_qp_common_test(e.IBV_QPT_UD, e.IBV_QPS_RTS, False, True,
AssertionError: PyverbsRDMAError not raised

======================================================================
FAIL: test_create_raw_qp_ex_with_illegal_caps_max_send_wr (tests.test_qp.QPTest)
Test Raw Packet QP creation via ibv_create_qp_ex with a QPAttr object with illegal max_send_wr.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/rdma-core/tests/test_qp.py", line 201, in test_create_raw_qp_ex_with_illegal_caps_max_send_wr
    self.create_qp_common_test(e.IBV_QPT_UD, e.IBV_QPS_RTS, False, True,
AssertionError: PyverbsRDMAError not raised

----------------------------------------------------------------------
Ran 26 tests in 0.020s
```

Reply:
Ah ok. Thanks! Seems like an overflow in irdma_get_sqdepth in the roundup function. Let me review it once and see what is the best path forward.

Reply:
Sorry for the delay!

The limits are stored in uk_attrs struct already and no need for the defines. So something like this should do it.

@@ -2129,6 +2130,8 @@ struct ibv_qp *irdma_ucreate_qp(struct ibv_pd *pd,
 #endif
        if (attr->cap.max_send_sge > uk_attrs->max_hw_wq_frags ||
            attr->cap.max_recv_sge > uk_attrs->max_hw_wq_frags ||
+           attr->cap.max_send_wr > uk_attrs->max_hw_wq_quanta ||
+           attr->cap.max_recv_wr > uk_attrs->max_hw_rq_quanta ||
            attr->cap.max_inline_data > uk_attrs->max_hw_inline) {
                errno = EINVAL;
                return NULL;

Reply:
Fixed